### ISSUE TITLE
Applying swaps

### ISSUE THREAD

**Issue by y-richie-y** [oxford-quantum-group/discopy#17]
Currently, if we want to apply a swap between wires `i` and `j`, we need to do define the relevant swaps and add identity wires to the left and right of the swaps, or come up with the exact permutation, both of which is not very clean.

Proposal: 
Implement `diagram.apply_swap(i, j)`, which applies swaps (and the relevant identities) to the `i`th and `j`th wires.

**Comment by toumix** [comment]
I'm not sure I like the method name `apply_swap` so much, maybe `diagram.permute` would sound better?

We could also overload the method so it can either take a pair of ints as @y-richie-y proposed, or a list of ints to apply an arbitrary permutation. This is related to issue https://github.com/oxford-quantum-group/discopy/issues/37.

**Comment by y-richie-y** [comment]
Addressed by 8bb77d4. Partial permutations are also accepted now by `monoidal.permute()`

```python
from discopy import *
n, s, p = map(Ty, "nsp")
Word("werd", n @ s @ p @ s @ n).permute(4, 0).draw()
```

<img width="611" alt="image" src="https://user-images.githubusercontent.com/13847804/158501630-f5142d85-23af-4270-b48a-d924702bb862.png">

```python
from discopy import *
n, s, p = map(Ty, "nsp")
Word("werd", n @ s @ p @ s @ n).permute(3, 2, 1).draw()
```

<img width="609" alt="image" src="https://user-images.githubusercontent.com/13847804/158501789-66bd42b7-f21f-4429-b3ab-99e29fd85212.png">
